src/chewingio.c

```diff
@@ -1979,6 +1979,128 @@ CHEWING_API int chewing_userphrase_get(ChewingContext *ctx,
 #endif
 }
 
+CHEWING_API int chewing_userphrase_get_freq(ChewingContext *ctx,
+                                       const char phrase_buf[], const char bopomofo_buf[],
+                                       unsigned int *orig_freq, unsigned int *max_freq, unsigned int *user_freq, unsigned int *time)
+{
+    ChewingData *pgdata = NULL;
+    
+    int ret;
+    int phone_len;
+    int phrase_len;
+    uint16_t phone_buf[MAX_PHONE_SEQ_LEN];
+#ifdef WITH_SQLITE3
+    int i;
+#else
+    const TreeType *tree_pos;
+    Phrase phrase;
+    UserPhraseData *uphrase;
+#endif
+
+    if (!ctx || !phrase_buf || !bopomofo_buf) {
+        return -1;
+    }
+
+    pgdata = ctx->data;
+    phrase_len = ueStrLen(phrase_buf);
+    phone_len = UintArrayFromBopomofo(NULL, 0, bopomofo_buf);
+
+    ret = UintArrayFromBopomofo(phone_buf, phone_len + 1, bopomofo_buf);
+    if (ret == -1) {
+        return 0;
+    }
+
+    if (phone_len != phrase_len) {
+        LOG_WARN("Do not update userphrase because phone_buf length %d != phrase_buf length %d", phone_len, phrase_len);
+        return USER_UPDATE_FAIL;
+    }
+
+    if (phrase_len > MAX_PHRASE_LEN) {
+        LOG_WARN("phrase_buf length %d > MAX_PHRASE_LEN (%d)", phrase_len, MAX_PHRASE_LEN);
+        return -1;
+    }
+
+#if WITH_SQLITE3
```

Comment:
Can you avoid manipulating SQLite here? Conceptually, we wish to maintain SQLite-specific functions in file `userphrase-sql.c` rather than every piece of libchewing source files.